File: data/sliding-window/max-points-from-cards.cpp
#include <vector>
#include <numeric>
#include <algorithm>

using namespace std;

int maxScore(vector<int>& cardPoints, int k) {
    int n = cardPoints.size();
    int windowSize = n - k;
    int currentSum = 0;
    
    for (int i = 0; i < windowSize; ++i) {
        currentSum += cardPoints[i];
    }
    
    int minSum = currentSum;
    int totalSum = accumulate(cardPoints.begin(), cardPoints.end(), 0);
    
    for (int i = windowSize; i < n; ++i) {
        currentSum += cardPoints[i] - cardPoints[i - windowSize];
        minSum = min(minSum, currentSum);
    }
    
    return totalSum - minSum;
}

File: data/sliding-window/max-points-from-cards.py
def maxScore(cardPoints: list[int], k: int) -> int:
    n = len(cardPoints)
    window_size = n - k
    current_sum = sum(cardPoints[:window_size])
    min_sum = current_sum
    total_sum = sum(cardPoints)
    
    for i in range(window_size, n):
        current_sum += cardPoints[i] - cardPoints[i - window_size]
        min_sum = min(min_sum, current_sum)
        
    return total_sum - min_sum
